src/compositions/useFlowRuns.ts

```diff
@@ -1,39 +1,50 @@
 import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
-import { computed, Ref, ref, unref } from 'vue'
+import { computed, Ref, ref, watch } from 'vue'
 import { useCan } from '@/compositions/useCan'
 import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+import { FlowRunsFilter } from '@/models/Filters'
 import { WorkspaceFlowRunsApi } from '@/services'
+import { flowRunStorage, useFlowRunStorage } from '@/services/storage'
 import { UseEntitySubscription } from '@/types/useEntitySubscription'
 
 export type UseFlowRuns = UseEntitySubscription<WorkspaceFlowRunsApi['getFlowRuns'], 'flowRuns'>
 
-export function useFlowRuns(flowRunIds: string[] | Ref<string[] | null | undefined>): UseFlowRuns {
+export function useFlowRuns(filter: FlowRunsFilter | Ref<FlowRunsFilter | null | undefined>): UseFlowRuns {
   const api = useWorkspaceApi()
   const can = useCan()
-  const flowRunIdsRef = ref(flowRunIds)
+  const filterRef = ref(filter)
 
-  const parameters = computed<Parameters<typeof api.flowRuns.getFlowRuns> | null>(() => {
-    const ids = unref(flowRunIdsRef)
-
-    if (!ids || ids.length === 0) {
+  const parameters = computed<[FlowRunsFilter] | null>(() => {
+    if (!filterRef.value) {
       return null
     }
 
     if (!can.read.flow_run) {
       return null
     }
 
-    return [
-      {
-        flowRuns: {
-          id: ids,
-        },
-      },
-    ]
+    return [filterRef.value]
   })
 
   const subscription = useSubscriptionWithDependencies(api.flowRuns.getFlowRuns, parameters)
-  const flowRuns = computed(() => subscription.response)
+
+  const storage = useFlowRunStorage()
+
+  watch(() => subscription.response, response => {
+    if (response) {
+      flowRunStorage.addAll(response)
+    }
+  })
+
+  const flowRuns = computed(() => {
+    if (subscription.response) {
+      const ids = subscription.response.map(flowRun => flowRun.id)
+
+      return storage.getAll(ids)
+    }
+
+    return undefined
+  })
```

Comment:
@znicholasbrown these are important implementation pieces. The flow run ids from the subscription are being used to get flow runs from storage. And `getAll` automatically calls a cleanup (unsubscribe) method each time you call it that cleans out previous values. So whenever the subscription changes, the storage will get cleaned up.